### DISCUSSION TITLE
Possibility to actively control a session running on core-daemon with a python script?

### DISCUSSION THREAD

**Discussion by elebug** [coreemu/core#639] [Q&A] [answered]
Hello guys,

I wish to run an automated simulation session with an actively changing topology that requires external control of the session.
Mainly to achieve the following with a py script:
0. creating and starting a session and connecting the GUI to it for visualization.
1. creating/removing/positioning MDR nodes in the canvas.
2. open consoles and run apps on each node
3. modifying connection properties such as introducing more lose when nodes are further away

Is this remotely possible? 

Thank you in advance

**Comment by gh0st42** [comment, marked as answer]
it is possible to use the gRPC client interface either via [python](https://github.com/coreemu/core/tree/master/daemon/core/api/grpc) as shipped with coreemu itself or by generating your own client code using the [gRPC definitions](https://github.com/coreemu/core/tree/master/daemon/proto/core/api/grpc).

I've successfully used this to make a [Rust lib](https://crates.io/crates/coreemu) for remote control of coreemu sessions.

**Reply by elebug** [reply]
I see... thanks for this info.
